contrib/coredns/policy/policy.go

```diff
@@ -119,6 +119,70 @@ func (p *PolicyMiddleware) getEDNS0Attrs(r *dns.Msg) ([]*pb.Attribute, bool) {
 	return attrs, foundSourceIP
 }
 
+type NewLocalResponseWriter struct {
+	localAddr  net.Addr
+	remoteAddr net.Addr
+	Msg        *dns.Msg
+}
+
+func (r *NewLocalResponseWriter) Write(b []byte) (int, error) {
+	r.Msg = new(dns.Msg)
+	return len(b), r.Msg.Unpack(b)
+}
+
+// These methods implement the dns.ResponseWriter interface from Go DNS.
+func (r *NewLocalResponseWriter) Close() error              { return nil }
+func (r *NewLocalResponseWriter) TsigStatus() error         { return nil }
+func (r *NewLocalResponseWriter) TsigTimersOnly(b bool)     { return }
+func (r *NewLocalResponseWriter) Hijack()                   { return }
+func (r *NewLocalResponseWriter) LocalAddr() net.Addr       { return r.localAddr }
+func (r *NewLocalResponseWriter) RemoteAddr() net.Addr      { return r.remoteAddr }
+func (r *NewLocalResponseWriter) WriteMsg(m *dns.Msg) error { r.Msg = m; return nil }
+
+// This function validates the response from middleware
+// Inputs are - context.Context, dns.ResponseWriter, *dns.Msg
+// Outputs are - int, error
+func (p *PolicyMiddleware) handlePermit(ctx context.Context, w dns.ResponseWriter, r *dns.Msg, attrs []*pb.Attribute) (int, error) {
+	lw := new(NewLocalResponseWriter)
+	status, err := middleware.NextOrFailure(p.Name(), p.Next, ctx, lw, r)
+	if err != nil {
+		return status, err
+	}
+
+	attrs[0].Value = "response"
+	for _, rr := range lw.Msg.Answer {
+		switch rr.Header().Rrtype {
+		case dns.TypeA, dns.TypeAAAA:
+			if a, ok := rr.(*dns.A); ok {
+				attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: a.A.String()})
+			} else if a, ok := rr.(*dns.AAAA); ok {
+				attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: a.AAAA.String()})
+			}
+		default:
+		}
+	}
+
+	// Dummy security policy
+	attrs = append(attrs, &pb.Attribute{Id: "policy_id", Type: "string", Value: "SECURITY_POLICY_1790"})
```

Comment:
You'll need to pass in the original Response as well, to get the policy ID. Or at least the policy ID itself.